contracts/modules/staking/StakingPool.sol

```diff
@@ -1250,9 +1250,36 @@ contract StakingPool is IStakingPool, ERC721 {
 
   /* pool management */
 
-  function setProducts(StakedProductParam[] memory params) external onlyManager {
-    uint[] memory productIds = new uint[](params.length);
+  function recalculateEffectiveWeights(uint[] calldata productIds) external {
+    (
+    uint globalCapacityRatio,
+    /* globalMinPriceRatio */,
+    /* initialPriceRatios */,
+    uint[] memory capacityReductionRatios
+    ) = ICover(coverContract).getPriceAndCapacityRatios(productIds);
+
+    uint _totalEffectiveWeight = totalEffectiveWeight;
+
+    for (uint i = 0; i < productIds.length; i++) {
+      uint productId = productIds[i];
+      StakedProduct memory _product = products[productId];
+
+      uint8 previousEffectiveWeight = _product.lastEffectiveWeight;
+      _product.lastEffectiveWeight = _getEffectiveWeight(
```

Comment:
As discussed, `lastEffectiveWeight` may overflow, we need a larger datatype + a check for overflows (because it may happen anyway). When an overflow is detected the type's max value instead.